indexer/packages/postgres/src/stores/fill-table.ts

```diff
@@ -156,11 +158,40 @@ export async function findAll(
     Ordering.DESC,
   );
 
-  if (limit !== undefined) {
+  if (limit !== undefined && page === undefined) {
     baseQuery = baseQuery.limit(limit);
   }
 
-  return baseQuery.returning('*');
+  /**
+   * If a query is made using a page number, then the limit property is used as 'page limit'
+   */
+  if (page !== undefined && limit !== undefined) {
+    /**
+     * We make sure that the page number is always >= 1
+     */
+    const currentPage = Math.max(1, page);
+    const offset = (currentPage - 1) * limit;
+
+    /**
+     * We need to remove the sorting as it is not necessary in this case.
+     * Also a casting of the ts type is required since the infer of the type
+     * obtained from the count is not performed.
+     */
+    const count = await baseQuery.clone().clearOrder().count({ count: '*' }).first() as unknown as { count?: string };
```

Comment:
nit: Please add types to all declared variables.

Reply:
Yeah, we "fixed" in 0d14f11d9d8046396db741056ad725881fca9699. 
Thank you!